File: UNIDADE/flutter_application_finalProject/lib/iot/RelatorioDiario.dart
class RelatorioDiario {
  final String dataRef; // "2026-05-01"
  final double tempMedia;
  final double tempMinima;
  final double tempMaxima;
  final double umidadeMedia;
  final double umidadeMinima;
  final double umidadeMaxima;
  final double soloMedia;
  final double soloMinima;
  final double soloMaxima;
  final int totalLeituras;

  RelatorioDiario({
    required this.dataRef,
    required this.tempMedia,
    required this.tempMinima,
    required this.tempMaxima,
    required this.umidadeMedia,
    required this.umidadeMinima,
    required this.umidadeMaxima,
    required this.soloMedia,
    required this.soloMinima,
    required this.soloMaxima,
    required this.totalLeituras,
  });

  factory RelatorioDiario.fromJson(Map<String, dynamic> json) {
    return RelatorioDiario(
      dataRef: (json['dataRef'] ?? '').toString(),
      tempMedia: (json['tempMedia'] ?? 0).toDouble(),
      tempMinima: (json['tempMinima'] ?? 0).toDouble(),
      tempMaxima: (json['tempMaxima'] ?? 0).toDouble(),
      umidadeMedia: (json['umidadeMedia'] ?? 0).toDouble(),
      umidadeMinima: (json['umidadeMinima'] ?? 0).toDouble(),
      umidadeMaxima: (json['umidadeMaxima'] ?? 0).toDouble(),
      soloMedia: (json['soloMedia'] ?? 0).toDouble(),
      soloMinima: (json['soloMinima'] ?? 0).toDouble(),
      soloMaxima: (json['soloMaxima'] ?? 0).toDouble(),
      totalLeituras: (json['totalLeituras'] ?? 0) as int,
    );
  }

  // Formato curto para eixo X do gráfico: "05/01"
  String get labelCurto {
    if (dataRef.length >= 10) {
      return dataRef.substring(5).replaceAll('-', '/');
    }
    return dataRef;
  }
}

File: UNIDADE/apiProjetoSensor/src/main/java/com/example/apiProjeto/model/RelatorioDiario.java
package com.example.apiProjeto.model;

import jakarta.persistence.*;
import java.time.LocalDate;
import java.time.LocalDateTime;

@Entity
@Table(name = "relatorio_diario", uniqueConstraints = @UniqueConstraint(columnNames = "data_ref"))
public class RelatorioDiario {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "data_ref", nullable = false, unique = true)
    private LocalDate dataRef;

    private double tempMedia;
    private double tempMinima;
    private double tempMaxima;

    private double umidadeMedia;
    private double umidadeMinima;
    private double umidadeMaxima;

    private double soloMedia;
    private double soloMinima;
    private double soloMaxima;

    private int totalLeituras;

    @Column(name = "atualizado_em", columnDefinition = "DATETIME DEFAULT CURRENT_TIMESTAMP")
    private LocalDateTime atualizadoEm;

    @PrePersist
    @PreUpdate
    public void preUpdate() {
        atualizadoEm = LocalDateTime.now();
    }

    public RelatorioDiario() {}

    public Long getId() { return id; }

    public LocalDate getDataRef() { return dataRef; }
    public void setDataRef(LocalDate dataRef) { this.dataRef = dataRef; }

    public double getTempMedia() { return tempMedia; }
    public void setTempMedia(double tempMedia) { this.tempMedia = tempMedia; }

    public double getTempMinima() { return tempMinima; }
    public void setTempMinima(double tempMinima) { this.tempMinima = tempMinima; }

    public double getTempMaxima() { return tempMaxima; }
    public void setTempMaxima(double tempMaxima) { this.tempMaxima = tempMaxima; }

    public double getUmidadeMedia() { return umidadeMedia; }
    public void setUmidadeMedia(double umidadeMedia) { this.umidadeMedia = umidadeMedia; }

    public double getUmidadeMinima() { return umidadeMinima; }
    public void setUmidadeMinima(double umidadeMinima) { this.umidadeMinima = umidadeMinima; }

    public double getUmidadeMaxima() { return umidadeMaxima; }
    public void setUmidadeMaxima(double umidadeMaxima) { this.umidadeMaxima = umidadeMaxima; }

    public double getSoloMedia() { return soloMedia; }
    public void setSoloMedia(double soloMedia) { this.soloMedia = soloMedia; }

    public double getSoloMinima() { return soloMinima; }
    public void setSoloMinima(double soloMinima) { this.soloMinima = soloMinima; }

    public double getSoloMaxima() { return soloMaxima; }
    public void setSoloMaxima(double soloMaxima) { this.soloMaxima = soloMaxima; }

    public int getTotalLeituras() { return totalLeituras; }
    public void setTotalLeituras(int totalLeituras) { this.totalLeituras = totalLeituras; }

    public LocalDateTime getAtualizadoEm() { return atualizadoEm; }
}
